src/solver/pad_constant/pad_constant_fwd_contiguous.cpp

```diff
@@ -0,0 +1,133 @@
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/datatype.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/kernel_build_params.hpp"
+#include "miopen/miopen.h"
+#include "miopen/mlo_internal.hpp"
+#include "miopen/pad_constant/solvers.hpp"
+#include "miopen/pad_constant/invoke_params.hpp"
+#include "miopen/tensor_view_5d.hpp"
+
+namespace miopen {
+namespace solver {
+namespace pad_constant_fwd_contiguous {
+bool PadConstantFwdContiguous::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    if(!problem.IsSameType())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+ConvSolution PadConstantFwdContiguous::GetSolution(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::pad_constant_fwd_contiguous::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+    auto ydims  = problem.GetYDesc().GetLengths();
+
+    auto input_dtype  = miopen::GetDataType(problem.GetXDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetYDesc().GetType());
+
+    // for xgridsize: 5d -> 1d
+    size_t output_size = problem.GetYDesc().GetElementSize();
+
+    size_t xlocalsize = 1024;
+    // AlignUp blows up, because output_size can be > int_max. Lovely.
+    // size_t xgridsize  = AlignUp(output_size, xlocalsize);
+    size_t xgridsize =
+        (((static_cast<size_t>(output_size) + xlocalsize - 1) / xlocalsize) * xlocalsize);
+    size_t ylocalsize = 1;
+    size_t ygridsize  = 1;
+
+    auto kernel = KernelInfo{};
+
+    kernel.kernel_file = "MIOpenPadConstantFwd.cpp";
+    kernel.kernel_name = "PadConstantFwdContiguous";
+
+    // TODO: Actually understand how to use this properly
+    const auto build_params = KernelBuildParameters{
+        {"INPUT_TYPE", input_dtype == "half" ? "half" : "float"},
+        {"OUTPUT_TYPE", output_dtype == "half" ? "half" : "float"},
+    };
+
+    kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+    kernel.l_wk.push_back(xlocalsize);
+    kernel.l_wk.push_back(ylocalsize);
+
+    kernel.g_wk.push_back(xgridsize);
+    kernel.g_wk.push_back(ygridsize);
+
+    result.construction_params.push_back(kernel);
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& invoke_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params =
+                invoke_params.CastTo<miopen::pad_constant_fwd_contiguous::InvokeParams>();
+
+            auto xdims    = params.xDesc->GetLengths();
+            auto ydims    = params.yDesc->GetLengths();
+            auto xstrides = params.xDesc->GetStrides();
+            auto ystrides = params.yDesc->GetStrides();
+
+            tensor_view_5d_t input_tv, output_tv;
+
+            for(size_t i = 0; i < 5; i++)
```

Comment:
You're almost done. Forget about assuming you're getting a 5D tensor. That's my mistake. To be more clear, even if a Smaller than 5D(2D or 3D) tensor is passed, the tensor view must be made 5D. Please refer to the get_inner_expanded_tv function in the getitem PR. Tensor view work is a bit difficult actually.

Reply:
I've ported `get_inner_expanded_tv` and switched to it for the 5d tensor